cmd/flightctl-alert-exporter/main.go

```diff
@@ -39,59 +70,97 @@ func main() {
 	log.SetLevel(logLvl)
 
 	tracerShutdown := tracing.InitTracer(log, cfg, "flightctl-alert-exporter")
-	defer func() {
-		if err := tracerShutdown(ctx); err != nil {
-			log.Fatalf("failed to shut down tracer: %v", err)
-		}
-	}()
+	if tracerShutdown != nil {
+		cleanupFuncs = append(cleanupFuncs, func() error {
+			log.Info("Shutting down tracer")
+			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+			defer cancel()
+			return tracerShutdown(ctx)
```

Comment:
this didnt need any extra ctx or timeout. when this defer fired its exactly when it should
```suggestion
defer func() {
		if err := tracerShutdown(ctx); err != nil {
			log.Error("failed to shut down tracer: %v", err)
		}
	}()
```